src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java

```diff
@@ -180,6 +180,37 @@ public ModelObject resolveContext(Object context) {
     }
 
     /**
+     * Checks if the current user has permission to create a credential.
+     *
+     * @param context     the context.
+     * @param includeUser whether they can use their own credentials store.
+     * @return {@code true} if they can create a permission.
+     * @since FIXME
+     */
+    @Restricted(NoExternalUse.class)
+    @SuppressWarnings("unused") // used via jelly
+    public boolean hasCreatePermission(ModelObject context, boolean includeUser) {
+        if (includeUser) {
+            User current = User.current();
+            if (current != null && current.hasPermission(CREATE)) {
+                return true;
+            }
+        }
+        if (context == null) {
+            StaplerRequest request = Stapler.getCurrentRequest();
+            if (request != null) {
+                context = request.findAncestorObject(ModelObject.class);
```

Comment:
And if that is null also, do we gracefully return false, or does `lookupStores` give an NPE?